sheepdog/tasks/transactions.py

```diff
@@ -66,26 +64,27 @@ def postprocess_transaction(db, txn_id):
         logger.debug(
             "Starting postprocessing steps for transaction: {}".format(txn_id))
 
-        for entity in txn.entities:
-            if entity.action.lower() in ['delete']:
-                continue
+        inherit_release_blocking(db, txn)
 
-            node_ids.append(entity.id)
-
-    inherit_release_blocking(db, node_ids)
 
-
-def inherit_release_blocking(graph, node_ids):
+def inherit_release_blocking(graph, txn):
     """
     Given a list of node_ids, propagate release blocking property to relevant
     children nodes
-
     :param graph: psqlgraph db connection instance
     :type graph: psqlgraph.PsqlGraphDriver
-    :param node_ids: a list of node_ids to process
-    :type node_ids: List[str]
+    :param txn: transaction log object
     """
+
+    node_ids = []
     with graph.session_scope():
+
+        for entity in txn.entities:
```

Comment:
Correct me if I'm wrong but this looks like can be re-written as follows.

```suggestion
        node_ids = [e for e in txn.entities if e.action.lower() != 'delete']
```
This also gets rid of the overhead of resizing a python list when the internal array list structure reaches capacity.